src/spans/awsSpan.js

```diff
@@ -156,6 +167,10 @@ export const getEndFunctionSpan = (functionSpan, handlerReturnValue) => {
     return_value,
     [EXECUTION_TAGS_KEY]: ExecutionTags.getTags(),
   });
+  if (error !== undefined) {
+    addEventToSpan(newSpan, true);
+    addEnvsToSpan(newSpan, true);
+  }
```

Comment:
Also here, try to build the new span in "one step" and not build and then extend